src/commands/aksCRUDDiagnostics/aksCRUDDiagnostics.ts

```diff
@@ -39,19 +39,27 @@ async function loadDetector(
     await longRunning(`Loading ${clustername} diagnostics.`,
           async () => {
             const clusterAppLensData = await getAppLensDetectorData(cloudTarget, "aks-category-crud");
-            const detectorMap = new Map();
+            if (clusterAppLensData === undefined) {
+              vscode.window.showErrorMessage('Unable to retrieve AppLens data.');
+              return;
+            }
 
             // Crud detector list is guranteed form the ARM call to aks-category-crud, under below data structure.
-            const crudDetectorList = clusterAppLensData?.properties.dataset[0].renderingProperties.detectorIds;
-
-            await Promise.all(crudDetectorList.map(async (detector: string) => {
-              const detectorAppLensData = await getAppLensDetectorData(cloudTarget, detector);
-              detectorMap.set(detector , detectorAppLensData);
-            }));
+            const crudDetectorList: string[] = clusterAppLensData?.properties.dataset[0].renderingProperties.detectorIds;
+            if (crudDetectorList.length === 0) {
```

Comment:
Again, moving this check earlier (to replace the `detectorMap.size` conditional later).

I've been burned in the past by AND'ing a bunch of conditionals, because you can't tell which of them failed. So my preference is generally to check each and fail early with an informative message for each.